File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # if array is empty return 0
        if len(nums) == 0:
            return 0
        # set up index counter 
        i = 0
        # loop starting at 1 with j comparing to left neighbor i at 0
        for j in range(1, len(nums)):
            # if the number at i is not equal to the number at j
            if nums[j] != nums[i]:
                # increment i
                i += 1
                # set the number at i to the number at j
                nums[i] = nums[j]
        return i + 1
        

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
//        if empty return 0
        if (nums.length == 0) return 0;
//        set up a counter index
        int i = 0;
//        loop through nums with j starting at 1
        for (int j = 1; j < nums.length; j++) {
//            if element on right and element on left arent equal
            if (nums[j] != nums[i]) {
//                increment index counter
                i++;
//                move index to right (j)
                nums[i] = nums[j];
            }
        }
        return i + 1;
    }
}